src/utils/proxy.js

```diff
@@ -171,6 +171,7 @@ const serveRedirect = async function ({ req, res, proxy, match, options }) {
         // Restore the URL if everything is correct
         if (presentedRoles.some((pr) => expectedRoles.has(pr))) {
           req.url = originalURL
+          match.force404 = false
```

Comment:
Not sure why `force404` is `true` when the role matches. This is returned from `netlify-redirector`.
Still looking into that one